### PR TITLE
Added solution

### PR DESCRIPTION
**Pull request by AndrewPolishchuc**

### CODE REVIEW

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 1]
```diff
@@ -1,127 +1,92 @@
...
-     *                   - 1-й элемент массива - год;
-     *                   - 2-й элемент массива - месяц;
-     *                   - 3-й элемент массива - день (число);
-     */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        try {
+            return Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
+        } catch (DateTimeException | ArrayIndexOutOfBoundsException exception) {
+            System.out.println(exception.getMessage());
```
Comment (andromash) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java:36]:
```suggestion
```

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 2]
```diff
@@ -1,127 +1,92 @@
...
-     * Дана произвольная дата getDate.
-     * Определите соотношение сегодня к someDate и верните строку:
-     * - "someDate is after текущая дата" - если getDate в будующем
-     * - "someDate is before текущая дата" - если getDate в прошлом
-     * - "someDate is today" - если someDate - сегодня
-     */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        LocalDate currentMoment = LocalDate.now();
+        int differenceInDates = currentMoment.compareTo(someDate);
```
Comment (andromash) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java:59]: Better use isAfter and isBefore methods

Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java:59]: Sure. Done.

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 1 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java:36]: You're right.

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 3]
```diff
@@ -1,127 +1,90 @@
...
-     *
-     * @param datePart Запрос на часть даты или всю дата целиком:
-     *                 - FULL - текущая дата целиком год, месяц, день (число месяца)
-     *                 в формате YYYY-MM-DD, возвращаемое значение по умолчанию;
-     *                 - YEAR - текущий год;
-     *                 - MONTH - название текущего месяца;
-     *                 - DAY - текущий день (число месяца);
-     *                 В любом другом случае бросить DateTimeException
-     **/
+    private static final DateTimeFormatter ENGLISH_DD_MM_YY_FORMAT
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java:15]:
```suggestion
    private static final DateTimeFormatter DATE_FORMATTER
```

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 4]
```diff
@@ -1,127 +1,90 @@
...
+            = DateTimeFormatter.ofPattern("dd MMM yyyy", Locale.ENGLISH);
+    private static final DateTimeFormatter ENGLISH_DD_MM_YY_MM_FORMAT
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java:17]:
```suggestion
    private static final DateTimeFormatter DATE_TIME_FORMATTER
```

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 5]
```diff
@@ -1,127 +1,90 @@
...
+            = DateTimeFormatter.ofPattern("dd MMMM yyyy HH:mm", Locale.ENGLISH);
+    private static final String PLUS_TWO_HOURS = "+02:00";
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java:19]:
```suggestion
    private static final String UKRAINE_OFFSET = "+02:00";
```

File: src/main/java/core/basesyntax/JavaDateTimeApi.java [hunk 6]
```diff
@@ -1,127 +1,90 @@
...
     }
 
-    /**
-     * Дана дата в строковом формате и временная зона.
-     * Верните LocalDateTime в этой временной зоне.
-     * @return LocalDateTime
-     */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        return ZonedDateTime.parse(dateInString + "[" + zone + "]").toLocalDateTime();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/JavaDateTimeApi.java:64]: let's try to be less straightforward and solve this with the help of Instant class

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 3 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java:15]: Done

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 4 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java:17]: Done

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 5 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java:19]: Done

File: src/main/java/core/basesyntax/JavaDateTimeApi.java (hunk 6 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/JavaDateTimeApi.java:64]: Done

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**